Capri.Services/SystemSettings/SystemSettingsSetter.cs

```diff
@@ -0,0 +1,46 @@
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Microsoft.AspNetCore.Hosting;
+
+namespace Capri.Services.SystemSettings
+{
+    public class SystemSettingsSetter : ISystemSettingsSetter
+    {
+        private readonly string _appSettingsFilePath;
+        public SystemSettingsSetter(IHostingEnvironment env)
+        {
+            var fileProvider = env.ContentRootFileProvider;
+            var appSettingsFileInfo = fileProvider.GetFileInfo("appsettings.json");
+            _appSettingsFilePath = appSettingsFileInfo.PhysicalPath;
+        }
+
+        public IServiceResult<int> SetMaxNumOfMasterProposalsPerPromoter(int number)
+        {
+            var jsonObj = JsonObjFromPath(_appSettingsFilePath);
+            jsonObj["SystemSettings"]["MaxNumOfMasterProposalsPerPromoter"] = number;
+            SaveJobjectToFile(jsonObj, _appSettingsFilePath);
+            return ServiceResult<int>.Success(number);
+        }
+
+        public IServiceResult<int> SetMaxNumOfBachelorProposalsPerPromoter(int number) {
+            var jsonObj = JsonObjFromPath(_appSettingsFilePath);
+            jsonObj["SystemSettings"]["MaxNumOfBachelorProposalsPerPromoter"] = number;
+            SaveJobjectToFile(jsonObj, _appSettingsFilePath);
+            return ServiceResult<int>.Success(number);
+        }
+
+        private JObject JsonObjFromPath(string path) 
+        {
+            var inputJsonString = File.ReadAllText(path);
+            dynamic jsonObj = JsonConvert.DeserializeObject(inputJsonString);
```

Comment:
Very bad! Serialize it to some specific class. Dynamic type affects code readability and also performance.